interfaces/builtin/docker.go

```diff
@@ -0,0 +1,185 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var dockerPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the Docker daemon. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# Allow sockets
+/{,var/}run/docker.sock rw,
+/{,var/}run/docker/     rw,
+/{,var/}run/docker/**   mrwklix,
+/{,var/}run/runc/       rw,
+/{,var/}run/runc/**     mrwklix,
```

Comment:
It probably _can_ be in `SNAP_DATA`, but would require patching Docker/runc to accomplish.  The way `runc` is designed, it's supposed to be safe for separate instances to share this directory (especially since `runc` doesn't have a "daemon", so every container invoked with `runc` is technically a new instance sharing this directory).